cli/src/pcluster/schemas/cluster_schema.py

```diff
@@ -1110,7 +1161,62 @@ def make_resource(self, data, **kwargs):
         return AwsBatchSettings(**data)
 
 
-class CloudFormationClusterInfrastructureSchema(BaseSchema):
+class ByosSupportedDistrosSchema(BaseSchema):
+    """Represent the schema for SupportedDistros in a BYOS plugin."""
+
+    x86_64 = fields.List(fields.Str(), metadata={"update_policy": UpdatePolicy.UNSUPPORTED})
+    arm64 = fields.List(fields.Str(), metadata={"update_policy": UpdatePolicy.UNSUPPORTED})
+
+    @post_load
+    def make_resource(self, data, **kwargs):
+        """Generate resource."""
+        return ByosSupportedDistros(**data)
+
+
+class ByosQueueConstraintsSchema(BaseSchema):
+    """Represent the schema for QueueConstraints in a BYOS plugin."""
+
+    max_count = fields.Int(metadata={"update_policy": UpdatePolicy.UNSUPPORTED})
+    max_subnets_count = fields.Int(metadata={"update_policy": UpdatePolicy.UNSUPPORTED})
+
+    @post_load
+    def make_resource(self, data, **kwargs):
+        """Generate resource."""
+        return ByosQueueConstraints(**data)
+
+
+class ByosComputeResourceConstraintsSchema(BaseSchema):
+    """Represent the schema for ComputeResourceConstraints in a BYOS plugin."""
+
+    max_count = fields.Int(metadata={"update_policy": UpdatePolicy.UNSUPPORTED})
+    max_instance_types_count = fields.Int(metadata={"update_policy": UpdatePolicy.UNSUPPORTED})
+
+    @post_load
+    def make_resource(self, data, **kwargs):
+        """Generate resource."""
+        return ByosComputeResourceConstraints(**data)
+
+
+class ByosRequirementsSchema(BaseSchema):
```

Comment:
since you defined the requirements schema, shall we link requirements below to this?